nautobot/core/tests/test_commands.py

```diff
@@ -0,0 +1,66 @@
+# TODO(jathan): This file MUST NOT be merged into Nautobot v2 (next).
+from io import StringIO
+
+from django.core.management import call_command
+from django.core.management.base import CommandError
+from django.contrib.contenttypes.models import ContentType
+
+from nautobot.utilities.testing import TestCase
+from nautobot.dcim.models import VirtualChassis
+from nautobot.extras.models import ConfigContext, ConfigContextSchema, ExportTemplate
+
+
+class PreMigrateCommandTest(TestCase):
+    """Test the `nautobot-server pre_migrate` command."""
+
+    def run_command(self, *args):
+        out = StringIO()
+        err = StringIO()
+        call_command(
+            "pre_migrate",
+            *args,
+            stdout=out,
+            stderr=err,
+        )
+
+        return (out.getvalue(), err.getvalue())
+
+    def test_success(self):
+        """Test that duplicate ConfigContext[Schema] objects result in a failure."""
+        out, err = self.run_command()
+
+        self.assertIn("All pre-migration checks passed.", out)
+        self.assertEqual("", err)
+
+    def test_configcontext_failure(self):
+        """Test that duplicate ConfigContext objects result in a failure."""
+        ConfigContext.objects.create(name="cc1", data={})
+        ConfigContext.objects.create(name="cc1", data={})
```

Comment:
A more realistic test here (as I'm pretty sure this would fail if you use `validated_save()` instead of `create()`) would be to define a GitRepository and do:

```python
ConfigContext(name="cc1", data={}).validated_save()
ConfigContext(name="cc1", data={}, owner=git_repo).validated_save()
```

Similar suggestion on the ConfigContextSchema and ExportTemplate tests.